internal/circuitbreaker/breaker.go

```diff
@@ -98,14 +98,30 @@ func (b *breaker) do(ctx context.Context, fn func(ctx context.Context) (val inte
 
 // isReady determines the breaker is ready or not.
 // If the current breaker state is "Closed" or "Half-Open", this function returns true.
-func (b *breaker) isReady() (ok bool) {
-	st := b.currentState()
-	return st == StateClosed || st == StateHalfOpen
+func (b *breaker) isReady() (st State, err error) {
+	st = b.currentState()
+	switch st {
+	case StateOpen:
+		return st, errors.ErrCircuitBreakerOpenState
+	case StateHalfOpen:
+
+		// For flow control in the "Half-Open" state. It is limited to 50%.
+		// If this modulo is used, 1/2 of the requests will be error.
+		total := b.count.Load().(*count).Total()
+		if total != 0 && total%2 == 0 {
+			return st, errors.ErrCircuitBreakerHalfOpenFlowLimitation
+		}
+	}
+	return st, nil
 }
 
 func (b *breaker) success() {
-	b.count.Load().(*count).onSuccess()
-	if st := b.currentState(); st == StateHalfOpen {
+	cnt := b.count.Load().(*count)
+	cnt.onSuccess()
+
+	// halfOpenErrShouldTrip.ShouldTrip is true when the sum of the number of successes and failures is greater than the b.minSamples and when the error rate is greater than the b.halfOpenErrRate.
+	// In other words, if the error rate is less than the b.halfOpenErrRate, it can be judged that the success rate is high, so this function change to the "Close" state from "Half-Open".
```

Comment:
```suggestion
	// In other words, if the error rate is less than the b.halfOpenErrRate, it can be judged that the success rate is high, so this function change to the "Close" state from "Half-Open".
```